common/utils/demangle.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef KOKKOSTOOLS_COMMON_UTILS_DEMANGLE_HPP
+#define KOKKOSTOOLS_COMMON_UTILS_DEMANGLE_HPP
+
+#include <memory>
+#include <string>
+
+#if defined(__GXX_ABI_VERSION)
+#define HAVE_GCC_ABI_DEMANGLE
+#endif
+
+#if defined(HAVE_GCC_ABI_DEMANGLE)
+#include <cxxabi.h>
+#endif  // HAVE_GCC_ABI_DEMANGLE
+
+namespace KokkosTools
+{
+
+//! Demangle @p mangledName implementation.
+inline auto demangleNameImpl(const char* mangled_name)
+{
+  int status = 0;
+
+  std::unique_ptr<char, decltype(&std::free)> demangled_name(
+    abi::__cxa_demangle(mangled_name, nullptr, nullptr, &status),
+    &std::free
+  );
+
+  if(status != 0) demangled_name.reset();
+
+  return demangled_name;
+}
+
+/**
+ * @brief Demangle @p mangledName.
+ *
+ * @note If the name was successfully demangled, @p mangledName
+ *       is freed and the returned pointer must be freed by the caller.
+ */
+inline char* demangleName(char* mangledName)
+{
+#if defined(HAVE_GCC_ABI_DEMANGLE)
+    auto demangled_name = demangleNameImpl(mangledName);
+
+    if(demangled_name != nullptr)
+    {
+        std::free(mangledName);
```

Comment:
?!!

Reply:
I know, it's ugly. But it's how it was implemented in `kp_space_time_stack.cpp` before...

I'll have a look to see if I can get rid of it.

Reply:
This design might fit the current implementation details of space-time-stack, but I don't feel like it is a good idea to elevate it as is for reuse.  I find it too error prone.

Reply:
I fully agree!

Reply:
Should we refactor a bit `KernelPerformanceInfo` such that it stores the kernel name as a `std::string` instead of a `char*`? It seems it would make this class a bit more readable...

Reply:
```c++
KernelPerformanceInfo(std::string kName, KernelExecutionType kernelType)
```
is the constructor. So it goes from a `std::string` to a `char*` for some reason (that is maybe outdated).

Reply:
@dalg24 I've mae `kernelName` a `std::string`, so I could delete the "ugly" `demangleName` :wink: